redfish-core/lib/assembly.hpp

```diff
@@ -247,6 +248,34 @@ inline void getReadyToRemoveOfTodBattery(
                         assemblyIndex));
 }
 
+void getAssemblyPresence(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const auto& serviceName, const auto& assembly,
+                         const auto& assemblyIndex)
+{
+    nlohmann::json& assemblyArray = asyncResp->res.jsonValue["Assemblies"];
+    nlohmann::json& assemblyData = assemblyArray.at(assemblyIndex);
```

Comment:
I think this json reference is better to be moved inside body of `getProperty()`.
L265:
```
-   nlohmann::json& assemblyData = assemblyArray.at(assemblyIndex);
```

Reply:
then how to have default value for Status at line 258?

Reply:
It can be set before calling getProperty().
I think your new code L258 should do it.